stdlib/public/TensorFlow/Tensor.swift

```diff
@@ -137,10 +137,11 @@ public extension Tensor {
     let tensor = toAccelerator()
     // If the tensor is to be sent from host to TPU, the shape is specified on
     // TF CPU first, before TF CPU sends the tensor to TPU.
-    return #tfop("Identity",
-                 tensor,
-                 __shapes: [shape],
-                 __device: "/device:CPU:0")
+    let ret: TensorHandle<Scalar> = #tfop("Identity",
+                                          tensor,
+                                          __shapes: [shape],
+                                          __device: "/device:CPU:0")
+    return Tensor<Scalar>(handle: ret)
```

Comment:
Another redundant type signature. I think it’ll be shorter (and free of type signatures) if you write #tfop inline:

return Tensor(handle: #tfop...)

Reply:
I'm not confident of the compiler infra in terms of type inferencing/type checking of #tfop, and would prefer to be more explicit here.

Reply:
I don’t think I got your point about type inference being not predictable, but this is not doing much type inference here, particularly not for #tfop. In fact, this is just doing the exact same amount of type inference for #tfop in either way of writing. 

In a generic type context (declaration or extension), any reference of the type name with out a type signature is automatically bound by the generic type parameters. In this case, all mentions of Tensor in here directly means Tensor<Scalar>.

Tensor has exactly one initializer taking a handle argument, and this would be TensorHandle<Float>. There’s no ambiguity, so the contextual type in the ‘handle:’ slot is always TensorHandle<Float>. Writing ret many times with explicit signature is unnecessary code bloat and complexity IMO. 

I’m fine if you strongly prefer writing two statements here in this PR, but I’ll appreciate it if you at least change the ‘Tensor<Scalar>‘ to just ‘Tensor’ like you did for the function below for consistency and simplicity.

Reply:
Yes, changed `Tensor<Scalar>` to `Tensor`. Thanks for the catch.

I'd prefer an explicit local variable `ret` here for now. If we ever have too many such occurrences in our hand-written stdlib code, I agree we should revisit this.